main/config.c

```diff
@@ -362,20 +390,65 @@ struct ast_config_include *ast_include_new(struct ast_config *conf, const char *
 	else
 		inc->included_file = ast_strdup(included_file);
 
-	inc->exec = is_exec;
-	if (is_exec)
-		inc->exec_file = ast_strdup(exec_file);
-
 	if (!inc->include_location_file
 		|| !inc->included_file
-		|| (is_exec && !inc->exec_file)) {
+		|| ((include_type == CONFIG_STATEMENT_EXEC) && !inc->exec_file)) {
 		ast_includes_destroy(inc);
 		return NULL;
 	}
 
+	inc->include_type = include_type;
+
+	make_fn(fn, sizeof(fn), inc->included_file, from_file);
+	switch (inc->include_type) {
+	case CONFIG_STATEMENT_INCLUDE :
+		inc_file = ast_calloc(1, sizeof(*inc_file) + strlen(fn) + 1);
+		if (inc_file) {
+			strcpy(inc_file->path, fn);	/* safe */
+			AST_LIST_INSERT_TAIL(&inc->included_files, inc_file, next);
+		}
+		break;
+	case CONFIG_STATEMENT_TRYINCLUDE :
```

Comment:
shouldn't be a space before the colons here and elsewhere

Reply:
Fixed